pymarsys/contact.py

```diff
@@ -32,3 +32,290 @@ def create(self, contact, key_id=None, source_id=None):
             self.endpoint,
             payload=payload
         )
+
+    def create_many(self, contacts, key_id=None):
+        """
+        Create many contacts from a list of dictionaries.
+        :param contacts:  a list of key-value pairs which uniquely identify
+        the contact fields which will be created for the contact (e.g. a key
+        can be the email field ID (3), and its value is the email address of
+        the specific contact).
+        :param key_id: Key which identifies the contacts.
+        :return: Dictionary with a list of the ids of the created contacts.
+        """
+        payload = {
+            'contacts': contacts,
+        }
+        if key_id:
+            payload['key_id'] = key_id
+
+        return self.connection.make_call(
+            'POST',
+            self.endpoint,
+            payload=payload
+        )
+
+    def query(self,
+              return_,
+              field_id_dict=None,
+              limit=None,
+              offset=None,
+              excludeempty=None):
+        """
+        Generate a list of contacts with values for a specific field. For
+        example, field_id 1 returns the first names of all contacts. Both the
+        values and the contact IDs are returned.
+        :param return_: Field ID used to generate the list.
+        :param field_id_dict: Key is the field_id, the value determines if a
+        contact should be returned or not, [field_value].
+        :param limit: Specifies the maximum number of contacts to return.
+        :param offset: Specifies an offset for pagination (like in SQL).
+        :param excludeempty: If set to true, then all contacts with a null
+        or empty value in the requested field are not returned. Any value
+        except for true will be interpreted as false.
+        :return: List of contacts.
+        """
+        query_endpoint = '{}/{}/'.format(self.endpoint, 'query')
+        params = {
+            'return': return_,
+        }
+
+        for param_name in ('limit', 'offset', 'excludeempty'):
+            param_value = locals()[param_name]
+            if param_value is not None:
+                params[param_name] = param_value
+
+        if field_id_dict is not None:
+            if len(field_id_dict) > 1:
```

Comment:
I had to make a choice, I thought having `{'key_id':  'key_value'}` was clearer than a tuple. Both are ugly, but I am willing to change it.